compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -138,41 +138,28 @@ public STNode parse(ParserRuleContext context) {
      */
     @Override
     public STNode resumeParsing(ParserRuleContext context, Object... args) {
-        // TODO: revisit the commented resume-points
+        return resumeTopLevelNodeParsing(context, args);
+    }
+
+    public STNode resumeOtherNodesParsing(ParserRuleContext context, Object... args) {
```

Comment:
Whats the size of this method after refactoring?

Should we separate out syntax tokens (semicolons, commas, etc) as well?

Reply:
I think that ll be a good idea, resumeothernodes is about 300 lines long.